jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerSessionContext.kt

```diff
@@ -162,135 +170,249 @@ data class CodeModernizerSessionContext(
+            //  This Should be updated to fast fail the transform if install failed.
+            val copyOutput = runCopyCommand(mvnw)
+            if (copyOutput.exitCode != 0) {
+                LOG.error { "$mvnw command output: $copyOutput" }
+                val error = "Maven Copy: The exitCode should be 0 while it was ${copyOutput.exitCode}"
                 CodetransformTelemetry.mvnBuildFailed(
                     codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                     codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
                     reason = error
                 )
-                return null
             } else {
-                LOG.warn { "mvnw executed successfully" }
+                LOG.info { "$mvnw executed successfully" }
                 shouldTryMvnCommand = false
             }
         } catch (e: ProcessNotCreatedException) {
-            val error = "./mvnw failed to execute as its likely not a unix machine"
+            val error = "$mvnw failed to execute as its likely not a unix machine"
+            buildlogBuilder.appendLine("$mvnw failed to execute: $e")
             CodetransformTelemetry.mvnBuildFailed(
                 codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                 codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
                 reason = error
             )
-            LOG.warn { error }
+            LOG.warn(e) { error }
         } catch (e: Exception) {
+            val error = "Unexpected exception when running $mvnw"
+            buildlogBuilder.appendLine("$mvnw failed to execute: $e")
             CodetransformTelemetry.mvnBuildFailed(
                 codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                 codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
-                reason = e.message
+                reason = error
             )
             when {
                 e.message?.contains("Cannot run program \"./mvnw\"") == true -> {} // noop
                 else -> throw e
             }
+            LOG.error(e) { error }
         }
 
         // 2. maybe execute maven wrapper command
         if (shouldTryMvnCommand) {
-            LOG.warn { "Executing mvn" }
+            LOG.info { "Executing mvn" }
             try {
-                val output = runCommand("mvn")
-                if (output.exitCode != 0) {
-                    LOG.error { "Maven command output:\n$output" }
-                    val error = "The exitCode should be 0 while it was ${output.exitCode}"
+                val installOutput = runInstallCommand("mvn")
+                if (installOutput.exitCode != 0) {
+                    LOG.error { "Maven command output: $installOutput" }
+                    val error = "Maven Install: The exitCode should be 0 while it was ${installOutput.exitCode}"
```

Comment:
Lets make sure VSCode has the same "Maven Install" error prefix.

Reply:
Will fix